src/Microsoft.VisualStudio.Editors/SettingsDesigner/SettingsSingleFileGeneratorBase.vb

```diff
@@ -589,7 +589,7 @@ Namespace Microsoft.VisualStudio.Editors.SettingsDesigner
                 fullTypeName = projectRootNamespace & "."
             End If
             
-            If defaultNamespace <> "" Then
+            If defaultNamespace <> "" And Not defaultNamespace.Equals(MyNamespaceName) Then ' defaultNamespace, if none exists, will come in thru wszDefaultNamespace as My. We don't want to duplicate it.
```

Comment:
I took a look at the blame for this. Looks like you added the if-statement above this one for `projectRootNamespace`. So, is `My` being duplicated because your new if-statement prior to this has `projectRootNamespace` set to `My`? Then this if-statement was also executing causing `My.My` for `fullTypeName`?

Reply:
Correct.

Reply:
Relevant snippet:
>  There was a line that checks whether default namespace is empty, and if so does not append it to the generated type. So my *assumption* when modifying the file was that meant that having no default namespace would mean that the default namespace passed into the generator would be “”

Reply:
Summoning a [Ballad by Paul Vick](http://www.panopticoncentral.net/2003/08/18/the-ballad-of-andalso-and-orelse/):

```suggestion
            If defaultNamespace <> "" AndAlso Not defaultNamespace.Equals(MyNamespaceName) Then ' defaultNamespace, if none exists, will come in thru wszDefaultNamespace as My. We don't want to duplicate it.
```

Reply:
Interesting. Will add